### PR TITLE
Int b 21671 db create re fsc multipliers

### PR DESCRIPTION
**Pull request by msaki-caci**
## Summary

We are adding FSC multipliers table that will be used in the calculations for the price of the FSC.

## Verification Steps for the Author

These are to be checked by the author.

- [x] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [x] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Test that the migrations work using `make db_dev_migrate`
2. Verify the new table with the populated columns: id, low_weight, high_weight, multiplier, created_at, and updated_at. (I used Dbeaver)

#### Any new migrations/schema changes:

- [x] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [x] Have been communicated to #g-database.
- [x] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

File: migrations/app/schema/20241115214553_create_re_fsc_multipliers_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE IF NOT EXISTS re_fsc_multipliers (
+    id          uuid             NOT NULL,
```
Comment (ryan-mchugh) [inline migrations/app/schema/20241115214553_create_re_fsc_multipliers_table.up.sql:2]: I'd imagine we will want a unique primary key on this

File: pkg/models/re_fsc_multipliers.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+	"github.com/gofrs/uuid"
+	"google.golang.org/genproto/googleapis/type/decimal"
+)
+
+type FscMultiplier struct {
+	ID         uuid.UUID        `json:"id" db:"id" rw:"r"`
+	LowWeight  int              `json:"low_weight" rw:"r"`
+	HighWeight int              `json:"high_weight" rw:"r"`
+	Multiplier *decimal.Decimal `json:"multiplier"`
```
Comment (ryan-mchugh) [inline pkg/models/re_fsc_multipliers.go:14]: does this change? It, it also needs `rw:"r"`

File: migrations/app/schema/20241119160420_inserts_re_fsc_multipliers.up.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO re_fsc_multipliers (id,low_weight,high_weight,multiplier,created_at,updated_at) VALUES
+	 ('e8053bda-e19e-4343-b858-04d691f1438b',0,5000,0.000417,'2024-11-19 12:59:06.276892-06','2024-11-19 12:59:06.276892-06'),
+	 ('5f2d402e-9c41-4034-bfea-46e699e2ed95',5001,10000,0.0006255,'2024-11-19 12:59:06.276892-06','2024-11-19 12:59:06.276892-06'),
+	 ('0faa2887-bf9e-4fb5-9d96-42bd79d963bd',10001,24000,0.000834,'2024-11-19 12:59:06.276892-06','2024-11-19 12:59:06.276892-06'),
+	 ('9bb374f7-12a2-4c51-8391-a691406b4c2c',24001,99999999,0.00139,'2024-11-19 12:59:06.276892-06','2024-11-19 12:59:06.276892-06');
```
Comment (danieljordan-caci) [inline migrations/app/schema/20241119160420_inserts_re_fsc_multipliers.up.sql:5]: Could you consolidate these into one migration file? With how small they are and they do have relevance, I think one migration is appropriate.

File: migrations/app/schema/20241115214553_create_re_fsc_multipliers_table.up.sql (hunk 1 above)
Comment (msaki-caci) [reply migrations/app/schema/20241115214553_create_re_fsc_multipliers_table.up.sql]: I think it should be good now.

File: pkg/models/re_fsc_multipliers.go (hunk 2 above)
Comment (msaki-caci) [reply pkg/models/re_fsc_multipliers.go]: Yeah good catch. I was receiving errors on this before and meant to change it back once I got the errors fixed but it slipped

File: migrations/app/schema/20241119160420_inserts_re_fsc_multipliers.up.sql (hunk 3 above)
Comment (msaki-caci) [reply migrations/app/schema/20241119160420_inserts_re_fsc_multipliers.up.sql]: Got it consolidated

File: pkg/models/re_fsc_multipliers.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+	"github.com/gofrs/uuid"
+	"google.golang.org/genproto/googleapis/type/decimal"
+)
+
+type FscMultiplier struct {
+	ID         uuid.UUID        `json:"id" db:"id" rw:"r"`
+	LowWeight  int              `json:"low_weight" rw:"r"`
+	HighWeight int              `json:"high_weight" rw:"r"`
+	Multiplier *decimal.Decimal `json:"multiplier"  rw:"r"`
```
Comment (danieljordan-caci) [inline pkg/models/re_fsc_multipliers.go:14]: This shouldn't be a pointer since it can't be `NULL`

Comment (msaki-caci) [reply pkg/models/re_fsc_multipliers.go:14]: When I remove the pointer I get the following error:
 ```TableName passes lock by value: github.com/transcom/mymove/pkg/models.FscMultiplier contains google.golang.org/genproto/googleapis/type/decimal.Decimal contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex```

Comment (danieljordan-caci) [reply pkg/models/re_fsc_multipliers.go:14]: Interesting. I'll let it through then. Looks like Go doesn't play nice with Mutex. Could you add a comment in there mentioning that so we can have it for reference?

File: migrations/app/schema/20241115214553_create_re_fsc_multipliers_table.up.sql [hunk 5]
```diff
@@ -0,0 +1,19 @@
+CREATE TABLE IF NOT EXISTS re_fsc_multipliers (
+    id          uuid             NOT NULL PRIMARY KEY,
+    low_weight  int              NOT NULL,
+    high_weight int              NOT NULL,
+    multiplier  decimal          NOT NULL,
+    created_at  timestamp        NOT NULL DEFAULT NOW(),
+    updated_at  timestamp        NOT NULL DEFAULT NOW()
+);
+
+COMMENT ON TABLE re_fsc_multipliers IS 'Stores data needed to needed to calculate FSC';
```
Comment (danieljordan-caci) [inline migrations/app/schema/20241115214553_create_re_fsc_multipliers_table.up.sql:10]: Little typo here with duplicate `needed`

Comment (msaki-caci) [reply migrations/app/schema/20241115214553_create_re_fsc_multipliers_table.up.sql]: Fixed typo here [ea4941f](https://github.com/transcom/mymove/pull/14231/commits/ea4941f119363fabdec664c9c41ae32cd73f75d3)

File: pkg/models/re_fsc_multipliers.go (hunk 4 above)
Comment (msaki-caci) [reply pkg/models/re_fsc_multipliers.go:15]: Got the comment added here [5507d81](https://github.com/transcom/mymove/pull/14231/commits/5507d81416d97d45874147d8ec0d2b77d667a432)

**Review by danieljordan-caci — APPROVED** [review]

**Review by ryan-mchugh — APPROVED** [review]

**Comment by msaki-caci** [comment]
@begrohmann, @brianmanley-caci, or @deandreJones can I get one of yall's review on this?

**Review by begrohmann — APPROVED** [review]

**Pull request merged**